.github/push-image.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+set -e
+
+if [ "$#" -ne 1 ]; then
+    echo "Pushes a container image to Docker Hub with tags"
+    echo 
+    echo "Usage: $0 PROJECT"
+    echo "Example: $0 dummy-api"
+    echo "Relies on the following environment variables:"
+    echo "- GITHUB_REF, GITHUB_SHA (GH Action default)"
+    echo "- DOCKER_USERNAME, DOCKER_PASSWORD"
+    echo "Uses PROJECT:latest as input image."
+    echo "Override this behaviour with IMAGE_TAG environment variable."
+    exit 1
+fi
+
+PROJECT="$1"
```

Comment:
I usually follow http://www.tldp.org/LDP/Bash-Beginners-Guide/html/sect_03_02.html (section 3.2.2) for variables local to a script, keeping them lowercase as opposed to environment/global variables kept uppercase. It looks awful but it does communicate the scope and tools like `shellcheck` can enforce definition of local variables leaving the global ones alone.

Reply:
Cool, I didn't know shellcheck.